Remove UnitPrice and rework Money (#1327)
Shopify/hydrogen: docs/components/primitive/money.md

| withoutCurrency? | <code>boolean</code> | Whether to remove the currency symbol from the output. |
| withoutTrailingZeros? | <code>boolean</code> | Whether to remove trailing zeros (fractional money) from the output. If there are no trailing zeros, then the fractional money amount remains. For example, `$640.00` turns into `$640`. `$640.42` turns into `$640.42`. |

> | data | <code>PartialDeep&#60;MoneyV2&#62;</code> | An object with fields that correspond to the Storefront API's [MoneyV2 object](https://shopify.dev/api/storefront/reference/common-objects/moneyv2). |

Reviewer: ```suggestion
| data | <code>PartialDeep&#60;MoneyV2&#62;</code> | An object with fields that correspond to the Storefront API's [MoneyV2 object](https://shopify.dev/api/storefront/latest/objects/moneyv2). |
```

Reviewer: Realize this isn't quite part of your PR, and this legacy link will redirect -- however, I think it's still worthwhile referencing the most up-to-date URL. 😄

> | measurement | <code>PartialDeep&#60;UnitPriceMeasurement&#62;</code> | A [UnitPriceMeasurement object](https://shopify.dev/api/storefront/reference/products/unitpricemeasurement). |

Reviewer: ```suggestion
| measurement | <code>PartialDeep&#60;UnitPriceMeasurement&#62;</code> | A [UnitPriceMeasurement object](https://shopify.dev/api/storefront/latest/objects/unitpricemeasurement). |
```

Reviewer: Same comment here.

> | measurementSeparator | <code>ReactNode</code> | When the `measurement` prop is used, `measurementSeparator` allows you to customize the separator between the money output and the measurement output. Defaults to `'/'` |

Reviewer: ```suggestion
| measurementSeparator | <code>ReactNode</code> | Customizes the separator between the money output and the measurement output. Used with the`measurement` prop. Defaults to `'/'`. |
```
